addons/sale_project_stock/models/stock_picking.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def button_validate(self):
+        res = super().button_validate()
+        if res is not True:
+            return res
+        project_id = self.project_id
```

Comment:
Since the variable will contain a recordset then we could name it like:
```suggestion
        project = self.project_id
```

also the user could not have access to the project and so we should probably add a sudo.

I will do some tests with a user without any access to project.

Reply:
We did not add the group "project.group_project_user" to the project_id field in the field definition, we preferred to set it on the XML view. So I don't think a sudo is necessary